portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -42,11 +42,17 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-
+    String maxCommentParam = request.getParameter("max-comments");
+    int maxNumComments = maxCommentParam == null ? Integer.MAX_VALUE : Integer.parseInt(maxCommentParam);
+    
     PreparedQuery results = datastore.prepare(query);
 
+    int i = 0;
     List<Comment> commentsList = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      if (i == maxNumComments) {
+        break;
+      }
```

Comment:
Let's make this a for loop where i < the min of either results.length or maxComments, then you don't have to break, it will just stop